src/dags/importscripts/import_sport.py

```diff
@@ -0,0 +1,163 @@
+import csv
+from hashlib import blake2s
+import json
+import logging
+from typing import List, Tuple, TypedDict, Generator, Type
+
+
+# define logger for output to console
+logger = logging.getLogger(__name__)
+
+
+def get_dialect(file: str) -> Type:
+    """determine dialect of file
+
+    Args:
+        file: the path/to/file.ext that will be used to determine it's dialect
+
+    Returns:
+        A subclass dialect of Sniffer
+
+    """
+    with open(file) as data:
+        return csv.Sniffer().sniff(data.readline())
+
+
+def get_header(file: str) -> List:
+    """Return header row of file
+
+    Args:
+        file: the path/to/file.ext that will be used to determine it's header
+
+    Returns:
+        a list of columns in string
+
+    """
+    dialect = get_dialect(file)
+    with open(file, "r") as src:
+        data = csv.reader(src, dialect=dialect)
+        return next(data)
+
+
+def make_hash(composite_values: List, digest_size: int = None) -> int:
+    """The blake2s algorithm is used to generate a single hased value for source
+    composite values that uniquely identify a row.
+    In case the source itself doesn't provide a single solid identification as key.
+
+    Args:
+        composite_values: a concatenated string of record values based on
+                        which columns unqiuely identifiy a record
+        digest_size: size to set the max bytes to use for the hash
+
+    Returns:
+        a hased value of the composite values in int
+
+    Note:
+        The default digest size is for now set to 3 bytes, which is equivant
+        to ~ 10 karakters long.
+        Because the id column is preferably of type int, the hased value is converted
+        from hex to int.
+    """
+    digest_size = 3 if not digest_size else digest_size
+    return int(
+        blake2s("|".join(composite_values).encode(), digest_size=digest_size).hexdigest(), base=16
```

Comment:
You are converting a digest in bytes to hex and then from hex to int. This can be done in one step:

```python
int.from_bytes(blake2s(...).digest())
```

Reply:
`digest()` returns the internal byte representation (no conversion). `hexdigest()` does do a conversion.

Reply:
Aangepast